### PR TITLE
feat: add plugin for serializing disabled interactions

### PR DESCRIPTION
**Pull request by BeksOmega**
### Description

Adds a plugin for serializing and deserializing the deletable/movable/editable attributes of blocks. This makes it so that we can remove these attributes from the new JSON serialization system, but devs can still serialize them if they desire. For more information see [this forum post](https://groups.google.com/g/blockly/c/eP9PXVfCaHs/m/S4rPmwnTAQAJ).

### Testing

Added tests for serializing and deserializing all combinations of editable, movable, and deletable.
Tested while linked to the latest version of project-cereal, and all tests pass.

### CODE REVIEW

**Comment by BeksOmega** [comment]
Note: Tests are only failing because the CI doesn't link to the correct version of Blockly.

File: plugins/serialize-disabled-interactions/package.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  ],
+  "devDependencies": {
+    "@blockly/dev-scripts": "^1.2.10",
+    "@blockly/dev-tools": "^2.5.6",
+    "blockly": "^6.20210701.0"
+  },
+  "peerDependencies": {
+    "blockly": "^6.20210701.0"
+  },
+  "publishConfig": {},
```
Comment (alschmiedt) [inline plugins/serialize-disabled-interactions/package.json]: Add publishConfig information:
```
  "publishConfig": {
    "access": "public",
    "registry": "https://wombat-dressing-room.appspot.com"
  },
```

File: plugins/serialize-disabled-interactions/package.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@blockly/plugin-serialize-disabled-interactions",
+  "version": "0.0.0",
```
Comment (alschmiedt) [inline plugins/serialize-disabled-interactions/package.json]: Update the version to 1.0.0

File: plugins/serialize-disabled-interactions/package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  "engines": {
+    "node": ">=8.17.0"
+  },
+  "browserslist": [
+    "defaults",
+    "IE 11",
+    "IE_Mob 11"
+  ]
+}
```
Comment (alschmiedt) [inline plugins/serialize-disabled-interactions/package.json]: Add a new line.

File: plugins/serialize-disabled-interactions/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
+# @blockly/plugin-serialize-disabled-interactions [![Built on Blockly](https://tinyurl.com/built-on-blockly)](https://github.com/google/blockly)
+
+A [Blockly](https://www.npmjs.com/package/blockly) plugin that for the JSON
+serialization system that serializes which are blocks are not deletable,
+movable, or editable.
```
Comment (alschmiedt) [inline plugins/serialize-disabled-interactions/README.md]: A Blockly plugin ~~that~~ for the JSON serialization system that serializes which ~~are~~ block are not deletable, movable, or editable.

File: plugins/serialize-disabled-interactions/package.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    "dist",
+    "src"
+  ],
+  "devDependencies": {
+    "@blockly/dev-scripts": "^1.2.10",
+    "@blockly/dev-tools": "^2.5.6",
+    "blockly": "^6.20210701.0"
+  },
+  "peerDependencies": {
+    "blockly": "^6.20210701.0"
```
Comment (alschmiedt) [inline plugins/serialize-disabled-interactions/package.json:49]: Update to `7.20211209.1`.

File: plugins/serialize-disabled-interactions/package.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@blockly/plugin-serialize-disabled-interactions",
```
Comment (alschmiedt) [inline plugins/serialize-disabled-interactions/package.json:2]: Could you add this to the [blockly-samples README](https://github.com/google/blockly-samples/tree/master/plugins#readme) and [index.md ](https://github.com/google/blockly-samples/blob/master/gh-pages/index.md)?

File: plugins/serialize-disabled-interactions/src/index.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+/**
+ * Plugin serializer for saving and loading disabled interactions associated
+ * with blocks.
+ * @implements {Blockly.serialization.IPluginSerializer}
+ */
+class DisabledInteractionsSerializer {
+  /**
+   * Constructs te disabled interactions serializer.
```
Comment (alschmiedt) [inline plugins/serialize-disabled-interactions/src/index.js]: *the

File: plugins/serialize-disabled-interactions/test/serializer_test.mocha.js [hunk 8]
```diff
@@ -0,0 +1,194 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
```
Comment (alschmiedt) [inline plugins/serialize-disabled-interactions/test/serializer_test.mocha.js]: 2021

File: plugins/serialize-disabled-interactions/test/serializer_test.mocha.js [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Unit tests for a strict connection checker that extends
+ * Blockly.ConnectionChecker. Based on blockly's connection_checker_tests.js.
+ * @author fenichel@google.com (Rachel Fenichel)
```
Comment (alschmiedt) [inline plugins/serialize-disabled-interactions/test/serializer_test.mocha.js]: Please update

**Review by alschmiedt — APPROVED** [review]
This looks great : ) There are mostly just small nits.

Sorry it took me so unbelievably long to review! 

Also the base branch should be updated to master. Since it is all in its own directory I don't think this should be too difficult.

**Review by alschmiedt — APPROVED** [review]

**Pull request merged**